### PR TITLE
Added doc for switching elasticsearch to infinity

### PR DESCRIPTION
**Pull request by yuzhichang**
### What problem does this PR solve?

Added doc for switching elasticsearch to infinity

### Type of change

- [x] New Feature (non-breaking change which adds functionality)
- [x] Documentation Update

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -268,6 +279,20 @@ python3 download_deps.py
 docker build -f Dockerfile -t infiniflow/ragflow:dev .
 ```
 
+## 🔧 Build a Docker image for linux arm64
```
Comment (JinHai-CN) [inline README.md]: Move this chapter to build docker image

File: README.md [hunk 2]
```diff
@@ -244,7 +245,17 @@ Updates to the above configurations require a reboot of all containers to take e
 > $ docker compose -f docker/docker-compose.yml up -d
 > ```
 
-## 🔧 Build a Docker image without embedding models
+### Switch ElasticSearch to Infinity as the chunk store
+
+RAGFlow uses ElasticSearch as the chunk store(fulltext and vector) by default. If you want to switch to [Infinity](https://github.com/infiniflow/infinity/), follow the steps below:
```
Comment (JinHai-CN) [inline README.md]: RAGFlow uses Elasticsearch full-text and vector store by default.

File: README.md [hunk 3]
```diff
@@ -48,8 +48,9 @@
 - 🔎 [System Architecture](#-system-architecture)
 - 🎬 [Get Started](#-get-started)
 - 🔧 [Configurations](#-configurations)
-- 🔧 [Build a docker image without embedding models](#-build-a-docker-image-without-embedding-models)
-- 🔧 [Build a docker image including embedding models](#-build-a-docker-image-including-embedding-models)
```
Comment (JinHai-CN) [inline README.md]: All build chapter move to: build docker image doc

Comment (JinHai-CN) [reply README.md]: Also impacts other languages README.

File: README.md (hunk 2 above)
Comment (yuzhichang) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (yuzhichang) [reply README.md]: done

File: README.md [hunk 4]
```diff
@@ -244,6 +244,26 @@ Updates to the above configurations require a reboot of all containers to take e
 > $ docker compose -f docker/docker-compose.yml up -d
 > ```
 
+### Switch doc engine from Elasticsearch to Infinity
+
+RAGFlow uses Elasticsearch for full-text and vector store by default. If you want to switch to [Infinity](https://github.com/infiniflow/infinity/), follow the steps below:
```
Comment (writinwaters) [inline README.md]:
```suggestion
RAGFlow uses Elasticsearch by default for storing full text and vectors. To switch to [Infinity](https://github.com/infiniflow/infinity/), follow these steps:
```

File: README.md [hunk 5]
```diff
@@ -244,6 +244,26 @@ Updates to the above configurations require a reboot of all containers to take e
...
+
+1. Stop all running containers:
+
+   ```bash
+   $ docker compose -f docker/docker-compose.yml down -v
+   ```
+
+2. Edit `docker/.env` and set `DOC_ENGINE` to `infinity`.
```
Comment (writinwaters) [inline README.md]:
```suggestion
2. In **docker/.env**, set `DOC_ENGINE` to `infinity`.
```

File: README.md [hunk 6]
```diff
@@ -244,6 +244,26 @@ Updates to the above configurations require a reboot of all containers to take e
...
+
+3. Start the containers:
+
+   ```bash
+   $ docker compose -f docker/docker-compose.yml up -d
+   ```
+
+Note that Infinity `linux/arm64` variants are not officially supported yet.
```
Comment (writinwaters) [inline README.md]:
```suggestion
> [!WARNING] 
> Switching to Infinity on a Linux/arm64 machine is not yet officially supported.
```

File: docker/README.md [hunk 7]
```diff
@@ -26,7 +26,7 @@ The [.env](./.env) file contains important environment variables for Docker.
   The version of Elasticsearch. Defaults to `8.11.3`
 - `ES_PORT`  
   The port used to expose the Elasticsearch service to the host machine, allowing **external** access to the service running inside the Docker container.  Defaults to `1200`.
-- `ELASTIC_PASSWORD`  
+- `ELASTIC_PASSWORD`
```
Comment (writinwaters) [inline docker/README.md]:
```suggestion
- `ELASTIC_PASSWORD`  
```
Please recover the two spaces.

File: docs/guides/develop/build_docker_image.mdx [hunk 8]
```diff
@@ -61,4 +61,21 @@ docker build -f Dockerfile -t infiniflow/ragflow:dev .
 ```
 
   </TabItem>
+  <TabItem value="linux/arm64">
+
+## 🔧 Build a Docker image for linux arm64
+
+We have difficulty on building multi-arch image with CI. So we will not publish arm64 image officially in the near future.  
+However you can build it by yourself on a `linux/arm64` host:
```
Comment (writinwaters) [inline docs/guides/develop/build_docker_image.mdx]:
```suggestion
However, you can build an image yourself on a linux/arm64 host machine:
```

File: docs/guides/develop/build_docker_image.mdx [hunk 9]
```diff
@@ -61,4 +61,21 @@ docker build -f Dockerfile -t infiniflow/ragflow:dev .
 ```
 
   </TabItem>
+  <TabItem value="linux/arm64">
+
+## 🔧 Build a Docker image for linux arm64
+
+We have difficulty on building multi-arch image with CI. So we will not publish arm64 image officially in the near future.
```
Comment (writinwaters) [inline docs/guides/develop/build_docker_image.mdx]:
```suggestion
We are currently unable to regularly build multi-arch images with CI and have no plans to publish arm64 images in the near future.  
```

**Review by writinwaters — CHANGES_REQUESTED** [review]

**Review by writinwaters — CHANGES_REQUESTED** [review]
Some minor changes required.

File: README.md (hunk 6 above)
Comment (yuzhichang) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (yuzhichang) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (yuzhichang) [reply README.md]: done

File: docker/README.md (hunk 7 above)
Comment (yuzhichang) [reply docker/README.md]: done

File: docs/guides/develop/build_docker_image.mdx (hunk 9 above)
Comment (yuzhichang) [reply docs/guides/develop/build_docker_image.mdx]: done

**Pull request merged**